### ISSUE TITLE
CNTK_203_Reinforcement Learning Loss Function documentation

### ISSUE THREAD

**Issue by srini1948** [Microsoft/CNTK#2229]
In the "cart-pole" reinforcement learning tutorial sample CNTK_203 for the policy gradient method a loss function is defined that has "log of squared error". For computing the gradient "loss.forward, loss.backward" method is used. Is there any reference or documentation that describes this approach - I suspect it is a numerical perturbation method?

**Comment by sayanpa** [comment]
This example is to illustrate how to use forward and backward functions for computing network parameters. This is a toy problem so I do not have any reference for you. We do have a deep RL framework under 

> CNTK\bindings\python\cntk\contrib\deepRL

**Comment by srini1948** [comment]
The loss function used in deepRL are standard "cross entropy softmax" obtained from CNTK.losses and the gradients have analytical solutions. In CNTK_203_Reinforcement Learning the "loss" function is not a part of CNTK.losses. I wanted to know how the gradient is computed by CNTK in such cases.

**Comment by sayanpa** [comment]
Here is documentation on how to create user defined functions: https://cntk.ai/pythondocs/extend.html#user-defined-functions

CNTK allows you to write your own loss functions using CNTK primitives or defining your own using the code described above.

**Comment by srini1948** [comment]
Thanks!

Sent from Mail for Windows 10

**Comment by sayanpa** [comment]
closing this based on the response above.
